packages/fxa-auth-db-mysql/db-server/test/backend/db_tests.js

```diff
@@ -2597,6 +2597,84 @@ module.exports = function (config, DB) {
+      const subs = [...Array(10)].map(() => 'sub-' + Math.random())
+      let account
+
+      beforeEach(() => {
+        account = createAccount()
+        return db.createAccount(account.uid, account)
+      })
+
+      it('should create a new subscription for account', () => {
+        return db.createAccountSubscription(account.uid, subs[0], 'prod0', Date.now())
+          .then(result => assert.deepEqual(result, {}))
+      })
+
+      it('should fail to create for unknown account', () => {
+        return db.createAccountSubscription('12312312312', subs[1], 'prod2', Date.now())
+          .then(assert.fail, err => assert.equal(err.errno, 116, 'not found'))
+      })
+
+      it('should fail to create duplicate subscription id', () => {
+        const account2 = createAccount()
+        return db.createAccount(account2.uid, account2)
+          .then(() => db.createAccountSubscription(account.uid, subs[2], 'prod2', Date.now()))
+          .then(() => db.createAccountSubscription(account2.uid, subs[2], 'prod3', Date.now()))
+          .then(assert.fail, err => assert.equal(err.errno, 101, 'conflict'))
+      })
+
+      const pickSet = (list, name) => new Set(list.map(x => x[name]))
+
+      it('should support fetching all subscriptions', () =>
+        Promise.all([
+          db.createAccountSubscription(account.uid, subs[3], 'prod4', Date.now()),
+          db.createAccountSubscription(account.uid, subs[4], 'prod5', Date.now()),
+          db.createAccountSubscription(account.uid, subs[5], 'prod6', Date.now()),
+        ])
+        .then(() => db.fetchAccountSubscriptions(account.uid))
+        .then(result => {
+          assert.equal(result.length, 3)
+          assert.deepEqual(
+            pickSet(result, 'subscriptionId'),
+            new Set([ subs[3], subs[4], subs[5] ])
+          )
+          assert.deepEqual(
+            pickSet(result, 'productName'),
+            new Set([ 'prod4', 'prod5', 'prod6' ])
+          )
+        })
+      )
+
+      it('should support deleting a subscription', () =>
+        Promise.all([
+          db.createAccountSubscription(account.uid, subs[6], 'prod4', Date.now()),
+          db.createAccountSubscription(account.uid, subs[7], 'prod5', Date.now()),
+          db.createAccountSubscription(account.uid, subs[8], 'prod6', Date.now()),
+        ])
+        .then(() => db.deleteAccountSubscription(account.uid, subs[7]))
+        .then(() => db.fetchAccountSubscriptions(account.uid))
+        .then(result => {
+          assert.equal(result.length, 2)
+          assert.deepEqual(
+            pickSet(result, 'subscriptionId'),
+            new Set([ subs[6], subs[8] ])
+          )
+          assert.deepEqual(
+            pickSet(result, 'productName'),
+            new Set([ 'prod4', 'prod6' ])
+          )
+        })
+      )
+
+      it('should support fetching one subscription', () =>
+        db.createAccountSubscription(account.uid, subs[9], 'prod7', Date.now())
```

Comment:
What happens if the subscriptionId does not exist for this account, and similarly, if it exists but is associated with another account?

Reply:
Adding tests to assert failure for non-existent subscription and subscription for a different user